src/Microsoft.Extensions.Caching.Memory/Interfaces/IMemoryCacheEvictionTrigger.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Extensions.Caching.Memory
+{
+    public interface IMemoryCacheEvictionTrigger : IDisposable
+    {
+        // TODO: doc comments
+        Func<bool> EvictionCallback { get; set; }
```

Comment:
`void Initialize(Func<bool> evictionCallback);` or make it a parameter to Resume